crates/reporting/tests/test_reporting.rs

```diff
@@ -11731,4 +11731,98 @@ All branches in an `if` must have the same type!
     denoted with .. - is that what you meant?
     "###
     );
+
+    test_report!(
+        multiple_list_patterns_start_and_end,
+        indoc!(
+            r#"
+            when [] is
+                [.., A, ..] -> ""
+            "#
+        ),
+    @r###"
+    ── MULTIPLE LIST REST PATTERNS ─────────────────────────── /code/proj/Main.roc ─
+
+    This list pattern match has multiple rest patterns:
+
+    5│          [.., A, ..] -> ""
+                        ^^
+
+    I only support compiling list patterns with one .. pattern! Can you
+    remove this additional one?
+
+    ── UNSAFE PATTERN ──────────────────────────────────────── /code/proj/Main.roc ─
+
+    This `when` does not cover all the possibilities:
+
+    4│>      when [] is
+    5│>          [.., A, ..] -> ""
+
+    Other possibilities include:
+
+        _
+
+    I would have to crash if I saw one of those! Add branches for them!
+    "###
+    );
+
+    test_report!(
+        multiple_list_patterns_in_a_row,
+        indoc!(
+            r#"
+            when [] is
+                [A, .., .., B] -> ""
+            "#
+        ),
+    @r###"
+    ── MULTIPLE LIST REST PATTERNS ─────────────────────────── /code/proj/Main.roc ─
+
+    This list pattern match has multiple rest patterns:
+
+    5│          [A, .., .., B] -> ""
+                        ^^
+
+    I only support compiling list patterns with one .. pattern! Can you
+    remove this additional one?
```

Comment:
There is an extension here where we can give a nicer error message, saying that the second rest pattern is redundant, and making this a warning instead of an error - however I don't actually think this is a case people will actively run into, so let's do that in a follow-up if we think it's worthwhile.